src/cursor/cur_file.c

```diff
@@ -283,7 +283,7 @@ __curfile_reset(WT_CURSOR *cursor)
      * guarding the call to cursor bound reset with the API_USER_ENTRY macro. Doing so prevents
      * internal API calls from resetting cursor bounds unintentionally, e.g. cursor->remove.
      */
-    if (API_USER_ENTRY(session))
+    if (WT_IS_HS(session->dhandle) || API_USER_ENTRY(session))
```

Comment:
I assume this is done to allow for the history store cursor's reset logic to still reset the bounds

Reply:
Yes, that's correct.